### PR TITLE
Add a clear() method and improve destructor() to LROrderedLinkedListSingle.h

### PR DESCRIPTION
**Pull request by WanpengQian**
Add a clear() method, improve destructor() cleanup.

### CODE REVIEW

File: CPP/lists/LROrderedLinkedListSingle.h [hunk 1]
```diff
@@ -333,6 +328,41 @@ class LROrderedLinkedListSingle {
...
+        if (_leftRight.load() == READS_ON_LEFT) {
+            _head->nextRight = _tail;
+            _leftRight.store(READS_ON_RIGHT);
+            toggleVersionAndWait();
+            node = _head->nextLeft;
+            while (node != _tail) {
+                next = node->nextLeft;
+                delete node;
+                node = next;
+            }
```
Comment (pramalhe) [inline CPP/lists/LROrderedLinkedListSingle.h:350]: What about
_head->nextLeft = _tail;
don't we need that too?

File: CPP/lists/LROrderedLinkedListSingle.h [hunk 2]
```diff
@@ -333,6 +328,41 @@ class LROrderedLinkedListSingle {
...
+            _head->nextLeft = _tail;
+            _leftRight.store(READS_ON_LEFT);
+            toggleVersionAndWait();
+            node = _head->nextRight;
+            while (node != _tail) {
+                next = node->nextRight;
+                delete node;
+                node = next;
+            }
+        }
```
Comment (pramalhe) [inline CPP/lists/LROrderedLinkedListSingle.h:364]: and 
_head->nextRight = _tail;
is missing here, right?

**Review by pramalhe — CHANGES_REQUESTED** [review]

File: CPP/lists/LROrderedLinkedListSingle.h (hunk 2 above)
Comment (WanpengQian) [reply CPP/lists/LROrderedLinkedListSingle.h:364]: Yeah, you are right. I miss this also.

**Comment by WanpengQian** [comment]
Add the missing pointer setting of _head node.

File: CPP/lists/LROrderedLinkedListSingle.h (hunk 1 above)
Comment (WanpengQian) [reply CPP/lists/LROrderedLinkedListSingle.h:350]: Yeah. I miss this.

**Comment by pramalhe** [comment]
Excellent, thank you so much for the fix!
If you're even around Lausanne (Switzerland), let me know, pizza's on me  ;)

**Comment by WanpengQian** [comment]
Sure, someday in the future! ;-)

**Pull request merged**